### PR TITLE
Accident service

### PR DESCRIPTION
**Pull request by lukaszmichalskii**

### CODE REVIEW

**Comment by lukaszmichalskii** [comment]
Manually tested in Postman, did not implement endpoint unit tests and mocking.

**Comment by lukaszmichalskii** [comment]
Add constrains on description limit and proper error message, current behavior:
test

File: src/main/java/com/formulaoneapi/service/AccidentService.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        if (!accidentRepository.existsById(name)) {
+            throw new NoSuchElementException(String.format("Accident type '%s' not found", name));
+        }
+
+        return accidentRepository.save(accident);
+    }
+
+    public Accident save(Accident accident) {
+        return accidentRepository.save(accident);
+    }
```
Comment (Mario-659) [inline src/main/java/com/formulaoneapi/service/AccidentService.java:43]: Post should only create new resource, not update. I think we should return 409 after POST with name already existing. Maybe add `assertAccidentDoesNoExist` method here. Take a look how I solved this in team service PR.

File: src/main/java/com/formulaoneapi/model/Accident.java [hunk 2]
```diff
@@ -5,13 +5,18 @@
...
 import javax.persistence.Id;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
 
 @Entity
 @Data
 public class Accident {
     @Id
+    @NotBlank(message = "Name cannot be blank")
     private String name;
```
Comment (Mario-659) [inline src/main/java/com/formulaoneapi/model/Accident.java:21]: - add `@Size` to id 
- separate validation from id to a group (see my teams PR), when we are doing PUT we pass name in path, not in body. Valid body in PUT should look like this:
```json
{
    "description": "More informative description"
}
```

File: src/main/java/com/formulaoneapi/service/AccidentService.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    public Accident get(String name) {
+        return accidentRepository
+                .findById(name)
+                .orElseThrow(
+                        () -> new NoSuchElementException(String.format("Accident type '%s' not found", name))
+                );
+    }
+
+    public void remove(String name) {
```
Comment (Mario-659) [inline src/main/java/com/formulaoneapi/service/AccidentService.java:29]: add `@Transactional`

**Review by Mario-659 — CHANGES_REQUESTED** [review]
That assertion and exception throw in DELETE is good. 

I added id validation group and `ElementAlreadyExists` exception in team service PR so maybe we could merge my PR, then you can rebase on main and add those small changes with my solutions?

**Comment by lukaszmichalskii** [comment]
Done, check if everything is OK and merge if so

**Review by Mario-659 — APPROVED** [review]

**Pull request merged**